modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/utils/columnResizedHandler.js

```diff
@@ -22,9 +22,15 @@ export function getColumn(pages, nestedIndexes = []) {
 	nestedIndexes.forEach((indexes) => {
 		const {columnIndex, pageIndex, rowIndex} = indexes;
 
+		let index = pageIndex;
+
+		if (pages.length - 1 < pageIndex) {
+			index = 0;
+		}
```

Comment:
- I suggest you to rename `index` to `normalizedPageIndex`, because the current name it too generic.
- Maybe would be good flip the comparison, in order to make it more clear. 
  Something like:
  - if the page index goes over the pages length we should set the index to zero, otherwise we should keep the current page index
  - `pageIndex >= pages.length ? 0 :  pageIndex`

```suggestion
		const normalizedPageIndex = pageIndex >= pages.length ? 0 :  pageIndex
```